### PR TITLE
Changelog for 0.12.0

### PR DESCRIPTION
**Pull request by rgrunber**
Signed-off-by: Roland Grunberg <rgrunber@redhat.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,40 +1,76 @@
 # Quarkus extension for MicroProfile Language Server Changelog
 
+## 0.12.0 (July 29, 2022)
+
+### Enhancements
+ * Support for `@TemplateGlobal` annotation. See [#605](https://github.com/redhat-developer/quarkus-ls/issues/605).
+ * Support for `textDocument/InlayHint` in Qute templates. See [#595](https://github.com/redhat-developer/quarkus-ls/issues/595).
+ * Delay revalidation of Java files in Qute language server. See [#666](https://github.com/redhat-developer/quarkus-ls/pull/666).
+ * Delay revalidation after edit and improve cancel checking. See [#659](https://github.com/redhat-developer/quarkus-ls/pull/659).
+ * Improve memory and peformance of Qute language server. See [#654](https://github.com/redhat-developer/quarkus-ls/issues/654).
```
Comment (datho7561) [inline CHANGELOG.md:10]: Maybe it would be worthwhile to put these under a "performance" section?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,40 +1,76 @@
...
+ * `QuarkusConfigPropertiesProvider` void return type check doesn't work. See [#650](https://github.com/redhat-developer/quarkus-ls/issues/650).
+ * Linked editing doesn't work if variable is used as a parameter into a section. See [#638](https://github.com/redhat-developer/quarkus-ls/pull/638).
+
+### Build
+ * Update Jenkinsfile to use Java 17. See [#669](https://github.com/redhat-developer/quarkus-ls/pull/669).
+ * Adapt to new version of m2e in JDT-LS. See [#668](https://github.com/redhat-developer/quarkus-ls/pull/668).
+ * Remove unnecessary 2019-06 release repository from target platform. See [#658](https://github.com/redhat-developer/quarkus-ls/pull/658).
+ * Updates sites are not deployed anymore to download.jboss.org. See [#623](https://github.com/redhat-developer/quarkus-ls/issues/623).
+ * P2 update sites (quarkus.jdt, qute.jdt) are not archived for new releases/tags. See [#617](https://github.com/redhat-developer/quarkus-ls/issues/617).
+ * Remove unnecessary Gson dependency in pom files.. See [#672](https://github.com/redhat-developer/quarkus-ls/pull/672).
```
Comment (datho7561) [inline CHANGELOG.md:32]: Remove the second period here

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,40 +1,76 @@
...
+ * Move to LSP4J 0.13.0. See [#644](https://github.com/redhat-developer/quarkus-ls/issues/644).
```
Comment (datho7561) [inline CHANGELOG.md:33]: It looks like we use lsp4j 0.14.0 now

**Review by datho7561 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (rgrunber) [reply CHANGELOG.md]: Yup, I'll do that. I ended up combining the first two entries into 1.

File: CHANGELOG.md (hunk 3 above)
Comment (rgrunber) [reply CHANGELOG.md:35]: Yup, this was just a typo. We just bumped it even more.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,40 +1,81 @@
 # Quarkus extension for MicroProfile Language Server Changelog
 
+## 0.12.0 (July 29, 2022)
```
Comment (datho7561) [inline CHANGELOG.md:3]: If we're releasing this as the same time as vscode-microprofile, we should update this date to the 25th

**Review by datho7561 — CHANGES_REQUESTED** [review]

**Review by datho7561 — APPROVED** [review]

**Pull request merged**